src/scripts/main.js

```diff
@@ -1,5 +1,47 @@
 'use strict';
 
 document.addEventListener('click', e => {
-  // write code here
+  const item = e.target;
+
+  if (!item.classList.contains('wall')) {
+    return;
+  }
+
+  const wall = document.querySelector('.wall');
+  const spider = document.querySelector('.spider');
+
+  const computedStyle = window.getComputedStyle(wall, null);
+  const wallWidth = parseInt(computedStyle.getPropertyValue('width'));
+  const wallHeight = parseInt(computedStyle.getPropertyValue('height'));
+  const borderWidthWall = parseInt(computedStyle
+    .getPropertyValue('border-width'));
+
+  let coordsY = 0;
+  let coordsX = 0;
+  const spiderCoordsMinY = `${wall
+    .offsetTop + borderWidthWall + spider.height / 2}`;
+  const spiderCoordsMaxY = `${wall
+    .offsetTop + borderWidthWall + wallHeight - spider.height / 2}`;
+  const spiderCoordsMinX = `${wall
+    .offsetLeft + borderWidthWall + spider.width / 2}`;
+  const spiderCoordsMaxX = `${wall
+    .offsetLeft + borderWidthWall + wallWidth - spider.width / 2}`;
+
+  (e.clientY <= spiderCoordsMinY)
+    ? coordsY = spiderCoordsMinY
+    : (e.clientY >= spiderCoordsMaxY)
+      ? coordsY = spiderCoordsMaxY
+      : coordsY = e.clientY;
+
+  (e.clientX <= spiderCoordsMinX)
+    ? coordsX = spiderCoordsMinX
+    : (e.clientX >= spiderCoordsMaxX)
+      ? coordsX = spiderCoordsMaxX
+      : coordsX = e.clientX;
```

Comment:
You repeating yourself two times here. You could create a function that takes wallWidth/wallHeight, spider.width/spider.height, spiderCoordsMinY/spiderCoordsMinX, spiderCoordsMaxY/spiderCoordsMaxX as arguments and return needed numbers. It would reduce the code and you won't repeat yourself